/*
  Inventory

  This API Section holds messages used for exchanging vehicle inventory data between device and backend.

  The version of the OpenAPI document: 3

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";
package com.bosch.bfb.protobuf.inventory.v3;
import public "device_to_backend_message_meta.proto";
import public "software_type.proto";

message InventoryReportEvent {

  DeviceToBackendMessageMeta meta = 3347973;

  VehicleData vehicle = 342069036;

  // list of inventory software nodes
  repeated SoftwareNode softwareNodes = 113881418;

  // list of inventory hardware nodes
  repeated HardwareNode hardwareNodes = 294825385;

  // list of associations between hardware and software nodes
  repeated Association associations = 508736532;

  // authenticity data
  repeated Security security = 412251969;

}

message Association {

  // ID of the source node
  string sourceId = 135714457;

  // ID of the target node
  string targetId = 441951604;

}


message HardwareNode {

  // Globally unique identifier of the hardware node
  string id = 3355;

  // Semantic version of the hardware node
  string version = 351608024;

  // Name of the hardware node
  string name = 3373707;

  // Is this node addressable?
  bool addressable = 303163572;

  // Solution specific key-value tuples
  repeated Tuple parameters = 458736106;

}

message Security {

  // IDs of all software nodes covered by this security block
  repeated string softwareNodeIds = 455210418;

  // Authenticity data for the software nodes
  string security = 412251969;

}

message Tuple {

  // key
  string key = 106079;

  // value
  string value = 111972721;

}

message SoftwareNode {

  // identifier of the software (e.g. \"IVS Car Software\", \"Modem Firmware\").
  string id = 3355;

  // version string (e.g. \"1.0.0\")
  string version = 351608024;

  // Name of the software node
  string name = 3373707;

  SoftwareType type = 3575610;

  // Solution specific key-value tuples
  repeated Tuple parameters = 458736106;

}


message VehicleData {

  // Vehicle identifier that uniquely identifies the vehicle. (VIN)
  string id = 3355;

  // Version of this schema
  string version = 351608024;

  // Name of this schema
  string name = 3373707;

  // Authenticity data
  bool security = 412251969;

  // Solution specific key-value tuples
  repeated Tuple parameters = 458736106;

}